### PR TITLE
Threads notification proof of concept (MSC3773)

### PR DESCRIPTION
**Pull request by gsouquet**
Related to https://github.com/matrix-org/matrix-spec-proposals/pull/3773
Includes https://github.com/vector-im/element-web/issues/22980
Fixes https://github.com/vector-im/element-web/issues/22981

---
This PR currently has none of the required changelog labels.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -1120,14 +1123,30 @@ export class Room extends TimelineReceipts<EmittedEvents, RoomEventHandlerMap> {
         return event;
     }
 
+    /**
+     * Get sum of threads and roon notification counts
+     * @param {String} type The type of notification count to get. default: 'total'
+     * @return {Number} The notification count.
+     */
+    public getTotalUnreadNotificationCount(type = NotificationCountType.Total): number {
+        return (this.getUnreadNotificationCount(type) ?? 0) +
```
Comment (t3chguy) [inline src/models/room.ts:1132]: `getUnreadNotificationCount` already does the `?? 0`

File: src/models/room.ts [hunk 2]
```diff
@@ -1120,14 +1123,30 @@ export class Room extends TimelineReceipts<EmittedEvents, RoomEventHandlerMap> {
...
+
     /**
      * Get one of the notification counts for this room
      * @param {String} type The type of notification count to get. default: 'total'
      * @return {Number} The notification count, or undefined if there is no count
      *                  for this type.
      */
     public getUnreadNotificationCount(type = NotificationCountType.Total): number | undefined {
-        return this.notificationCounts[type];
+        return this.notificationCounts[type] ?? 0;
```
Comment (t3chguy) [inline src/models/room.ts:1149]: This behaviour & signature don't match, and this is a breaking change of the contract

File: src/sync.ts [hunk 3]
```diff
@@ -996,8 +997,8 @@ export class SyncApi {
         }
 
         const qps: ISyncParams = {
-            filter: filterId,
-            timeout: pollTimeout,
+            "filter": filterId,
+            "timeout": pollTimeout,
```
Comment (t3chguy) [inline src/sync.ts:1001]: any reason for this?

File: src/filter.ts [hunk 4]
```diff
@@ -220,6 +221,14 @@ export class Filter {
         setProp(this.definition, "room.timeline.limit", limit);
     }
 
+    /**
+     * Enable threads unread notification
+     * @param {boolean} enabled
+     */
+    public setUnreadThreadNotifications(enabled: boolean) {
+        setProp(this.definition, "room.timeline.unread_thread_notifications", enabled);
```
Comment (t3chguy) [inline src/filter.ts:229]: `org.matrix.msc3773.unread_thread_notifications`

**Comment by gsouquet** [comment]
Marking all of @t3chguy comments as "resolved" as this is a proof of concept and this code is not intended to reach production ever. This is to validate product behaviour

**Comment by gsouquet** [comment]
Is now on develop, https://github.com/matrix-org/matrix-react-sdk/pull/9400

**Pull request closed without merging**